backend/api/communication/src/main/java/co/airy/core/api/communication/Stores.java

```diff
@@ -212,6 +218,32 @@ public void deleteMetadata(Subject subject, String key) throws ExecutionExceptio
         producer.send(new ProducerRecord<>(applicationCommunicationMetadata, getId(subject, key).toString(), null)).get();
     }
 
+    public MetadataMap getMetadata(String subjectId) {
+        final ReadOnlyKeyValueStore<String, MetadataMap> metadataStore = getMetadataStore();
+        return metadataStore.get(subjectId);
+    }
+
+    public List<Conversation> addChannelMetadata(List<Conversation> conversations) {
```

Comment:
@paulodiniz and @lucapette So we are avoiding a join on the channels metadata and instead look it up on the fly.

Best case would be that we can also get rid of the channel join, but then it will become much harder to build the lucene index